cmd/juju/romulus/allocate/allocate.go

```diff
@@ -1,100 +1,88 @@
 // Copyright 2016 Canonical Ltd.
 // Licensed under the AGPLv3, see LICENCE file for details.
 
+// Package allocate defines the command used to update allocations.
 package allocate
 
 import (
 	"fmt"
-	"regexp"
-	"strings"
+	"strconv"
 
 	"github.com/juju/cmd"
 	"github.com/juju/errors"
-	"github.com/juju/gnuflag"
 	api "github.com/juju/romulus/api/budget"
-	"github.com/juju/utils"
 	"gopkg.in/macaroon-bakery.v1/httpbakery"
 
 	"github.com/juju/juju/cmd/modelcmd"
 )
 
-var budgetWithLimitRe = regexp.MustCompile(`^[a-zA-Z0-9\-]+:[0-9]+$`)
-
-type allocateCommand struct {
+type updateAllocationCommand struct {
 	modelcmd.ModelCommandBase
-	api       apiClient
-	Budget    string
-	ModelUUID string
-	Services  []string
-	Limit     string
+	api   apiClient
+	Value string
 }
 
-// NewAllocateCommand returns a new allocateCommand
-func NewAllocateCommand() modelcmd.ModelCommand {
-	return &allocateCommand{}
+// NewUpdateAllocationCommand returns a new updateAllocationCommand.
+func NewUpdateAllocationCommand() modelcmd.ModelCommand {
+	return &updateAllocationCommand{}
 }
 
-const doc = `
-Allocate budget for the specified applications, replacing any prior allocations
-made for the specified applications.
+func (c *updateAllocationCommand) newAPIClient(bakery *httpbakery.Client) (apiClient, error) {
+	if c.api != nil {
+		return c.api, nil
+	}
+	c.api = api.NewClient(bakery)
+	return c.api, nil
+}
 
-Examples:
-    # Assigns application "db" to an allocation on budget "somebudget" with
-    # the limit "42".
-    juju allocate somebudget:42 db
+type apiClient interface {
+	UpdateAllocation(string, string) (string, error)
+}
 
-    # Application names assume the current selected model, unless otherwise
-    # specified with:
-    juju allocate -m [<controller name:]<model name> ...
+const doc = `
+Updates an existing allocation for a model.
 
-    # Models may also be referenced by UUID when necessary:
-     juju allocate --model-uuid <uuid> ...
+Examples:
+    # Sets the allocation for the current model to 10.
+    juju update-allocation 10
 `
 
-// SetFlags implements cmd.Command.SetFlags.
-func (c *allocateCommand) SetFlags(f *gnuflag.FlagSet) {
-	c.ModelCommandBase.SetFlags(f)
-	f.StringVar(&c.ModelUUID, "model-uuid", "", "Model UUID of allocation")
-}
-
 // Info implements cmd.Command.Info.
-func (c *allocateCommand) Info() *cmd.Info {
+func (c *updateAllocationCommand) Info() *cmd.Info {
 	return &cmd.Info{
 		Name:    "allocate",
-		Args:    "<budget>:<value> <application> [<application2> ...]",
-		Purpose: "Allocate budget to applications.",
+		Args:    "<value>",
```

Comment:
do we not allow changing the budget as well? Moving an allocation from one budget to another? we used to have <budget>:<value> as an argument...